krabcake/rs_hello/src/lib.rs

```diff
@@ -1113,28 +1142,27 @@ pub extern "C" fn rs_shadow_load(addr: vg_addr, s1: vg_long) -> vg_long {
 
         if let Some(event) = STACKED_BORROW_EVENT {
             if event.stash == addr as vg_addr {
-                let s0 = event.tag.0;
+                let tag = event.tag;
+                let tag_s = tag.s();
                 msg!(
-                    b"rs_shadow_load 0x%08llx setting hack value for stashed tag %d\n\0",
+                    b"rs_shadow_load 0x%08llx setting hack value for stashed tag %s\n\0",
                     addr,
-                    s0
+                    tag_s.as_ptr(),
                 );
                 STACKED_BORROW_EVENT = None;
-                memory_shadow_value_hack = s0 as vg_long;
+                memory_shadow_value_hack = tag.to_vg_long();
             } else {
                 memory_shadow_value_hack = 0;
             }
         } else {
             memory_shadow_value_hack = 0;
         }
 
-        if s1 != 0 {
-            check_use_1(addr, s1 as u64);
-        }
+        check_use_1(addr, s1 as u64);
```

Comment:
Or.. this other one might be what is triggering the panic @pnkfelix

Reply:
(same as earlier comment; vgaddr 0 is not the right encoding for bottom; I wasn't thinking about it the right way when we spoke about this code earlier.)